c-sharp/DataStructures/DataStructures/LinkedList.cs

```diff
@@ -72,6 +72,24 @@ public void InsertAfter(int value, int newValue)
         }
           current = current.Next;
         }
+    }
+
+    public int FindKthValue(int value)
+    {
+      Node current = Head;
+      current.Next = current;
+      int length = 0;
+
+      while(current != null)
+      {
+        for(int i = 0; i < length - value; i++)
```

Comment:
This `for` loop never actually does anything because `length` is never not zero. Just drop the loop and `length++` right inside the `while` loop.